services/worker-manager/src/provisioner.js

```diff
@@ -96,26 +104,79 @@ class Provisioner {
    * Run a single provisioning iteration
    */
   async provision() {
+    if (this.provisioningLoopAlive) {
+      this.monitor.notice('loop-interference', {});
+      return;
+    }
+    this.provisioningLoopAlive = true;
```

Comment:
You'll probably want the rest of this function in a try/finally, otherwise this value may get set to `false` permanently.

Reply:
Ah, totally! Good catch